Sync/SyncStateMachine.swift

```diff
@@ -509,7 +509,7 @@ open class ServerConfigurationRequiredError: RecoverableSyncState {
         if let oldEngineConfiguration = s.engineConfiguration {
             metaGlobal = createMetaGlobalWithEngineConfiguration(oldEngineConfiguration, enginesEnablements: enginesEnablements)
         } else {
-            metaGlobal = createMetaGlobal(enginesEnablements: s.enginesEnablements)
```

Comment:
When constructing `s` above, we call `clearEnginesEnablements`, so AFAICT the value of `s.enginesEnablements` will always be null here. I expect we actually want to pass in the value from `oldScratchpad`, and then clear it in the new scratchpad `s` to indicate that the enablements have been dealt with.